x/rollup/module.go

```diff
@@ -46,13 +46,27 @@ type ModuleOutputs struct {
 }
 
 func init() { //nolint:gochecknoinits
-	appmodule.Register(&modulev1.Module{}, appmodule.Provide(ProvideModule), appmodule.Provide(ProvideCustomGetSigner))
+	appmodule.Register(
+		&modulev1.Module{},
+		appmodule.Provide(ProvideModule),
+		appmodule.Provide(ProvideApplyUserDepositGetSigner),
+		appmodule.Provide(ProvideSetL1AttributesGetSigner),
```

Comment:
How does this prevent arbitrary users from sending `ApplyUserDeposit` and `SetL1Attributes` messages? Where is the check done to make sure that these transactions are only coming from the OP Stack instead of an L2 user?

Reply:
It doesn't. Transactions come from the consensus layer via `FinalizeBlock`. Thus, it's Monomer's responsibility to ensure these messages never get sent to the mempool.

However, your comment makes me realize that the existing [check](https://github.com/polymerdao/monomer/blob/720fe686e4949fca49990538727da11b761f276b/mempool/mempool.go#L45) in the mempool is insufficient. Will work on fixing it after this PR.